### ISSUE TITLE
Why is spellright triggering in files I don't want spellcheck with?

### ISSUE THREAD

**Issue by nicholas-devlin** [bartosz-antosik/vscode-spellright#222]
.yaml file example: https://cl.ly/6e9e25459c7e

It also happens on and off for my .log files as well. I am using the latest developer release of VS code, and it's been happening for about a month. 

settings file changes from default spellright settings:

```json
    "spellright.language": [
        "en"
    ],
    "spellright.documentTypes": [
        "markdown",
        "latex",
        "plaintext"
    ],
    "spellright.notificationClass": "warning"
```

log and yaml files both say log and yaml in the file type indicator so I don't think they are like getting mislabeled as plaintext or anything.

**Comment by bartosz-antosik** [comment]
Could you please show a screenshot that would include the state of the switch in status bar? Around the "eye" symbol, far right side.

![image](https://user-images.githubusercontent.com/20492548/47112279-2d126200-d256-11e8-8177-e8c813400c76.png)

**Comment by bartosz-antosik** [comment]
Where is the configuration you show taken from?

**Comment by nicholas-devlin** [comment]
I was copying the config from my user settings: https://cl.ly/7221b41cfd93

I restarted and updated to the latest vs code build and it's working correctly again. It's been on and off for a while. 

If it happens again is there any other things I should include that would be helpful besides the file type info in the bottom right of the GUI?

Or do you only want to have issues raised in the the official vs code release and don't worry about insiders builds.

**Comment by bartosz-antosik** [comment]
@Geobert Could you please sketch a short recipe to reproduce this?

For example 1) Start VSCode (insiders?) then 2) do this 3) do that...

Does it happen only on first opening of the document?

**Comment by Geobert** [comment]
With the configuration above,
1/ start insiders
2/ F1, Open Settings (JSON)
3/ see spellright annotations
4/ switch to another tab
5/ switch back, annotations disappeared

**Comment by Geobert** [comment]
> Does it happen only on first opening of the document?

With a restored session, the first tab seems ok, switching to another opened tab shows unwanted spelling, switching away and back cleared them

**Comment by bartosz-antosik** [comment]
Great! Thank you. Will investigate this!

**Comment by Geobert** [comment]
Good luck!

**Comment by bartosz-antosik** [comment]
Oh Gosh. It's only happening in "Insiders" build. Have you had it (do you use at all) in regular release?

**Comment by Geobert** [comment]
I've copied all insiders config/extension/state into code equivalent, downloaded and installed 1.28.2. No bug here :-/

**Comment by nicholas-devlin** [comment]
Ah, nice find on the reproduction @Geobert I kept seeing it over past few days and couldn't find a way to reproduce it stably in order to take a picture lol. Here's a gif of me opening a JSON file for the first time and then the spell check getting applied after it loads:

https://cl.ly/e6fea493dc15

a gist of my entire settings file in case it matters somehow: https://gist.github.com/nicholas-devlin/745297dbc463daf5d51b212934682fb0

let me know if you need anything else, but it looks like you are able to reproduce as well, so should be good.

**Comment by nicholas-devlin** [comment]
When I remove plaintext I have not been able to reproduce this issue.

Also the only way I can reproduce now is by closing a window, then when the next file shows up it has the spell checking applied, when I actually click on a window it works correctly.

Here is a gif of me turning on and off the plaintext option:
https://cl.ly/e384bac076ee

**Comment by bartosz-antosik** [comment]
I have reported the problem as an VSCode's issue [#61441](https://github.com/Microsoft/vscode/issues/61441).

**Comment by bartosz-antosik** [comment]
The issue should be resolved in just published version 3.0.6. Please let me know if it works for you!

**Comment by Geobert** [comment]
It's fixed for me, thank you very much!
